platform/database/sandbox.go

```diff
@@ -0,0 +1,142 @@
+package database
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+
+	"github.com/GuiaBolso/darwin"
+	"github.com/jmoiron/sqlx"
+)
+
+type sandboxDB struct {
+	db *sqlx.DB
+	tx *sqlx.Tx
+}
+
+type sandboxTx struct {
+	tx                    *sqlx.Tx
+	rollBackedOrCommitted bool
+}
+
+// SandboxConnect returns a sandboxed database connection from the given configuration
+func SandboxConnect(config Config) (DB, error) {
+	db, err := connect(config)
+	if err != nil {
+		return nil, fmt.Errorf("can't connect to database: %v", err)
+	}
+
+	tx, err := db.Beginx()
+	if err != nil {
+		return nil, fmt.Errorf("could not create the test database transaction: %w", err)
+	}
+
+	return &sandboxDB{
+		db: db,
+		tx: tx,
+	}, nil
+}
+
+func (db *sandboxDB) Close() error {
+	err := db.tx.Rollback()
+	closeErr := db.db.Close()
+	if err != nil {
+		return fmt.Errorf("unable to rollback test transaction: %w", err)
+	}
+	if closeErr != nil {
+		return fmt.Errorf("unable to close database connection: %w", closeErr)
+	}
+	return nil
+}
+
+func (db *sandboxDB) NewGenericDriver(dialect darwin.Dialect) *darwin.GenericDriver {
+	panic("test database cannot produce a darwin driver")
+}
+
+func (db *sandboxDB) Begin() (Tx, error) {
+	db.tx.MustExec("SAVEPOINT go_pkg_database_sandbox_savepoint;")
+
+	return &sandboxTx{tx: db.tx}, nil
+}
+
+func (db *sandboxDB) SelectContext(ctx context.Context, dest interface{}, statement string, args ...interface{}) error {
+
+	return db.tx.SelectContext(ctx, dest, statement, args...)
+
+}
+
+func (db *sandboxDB) GetContext(ctx context.Context, dest interface{}, statement string, args ...interface{}) error {
+
+	return db.tx.GetContext(ctx, dest, statement, args...)
+
+}
+
+func (db *sandboxDB) ExecContext(ctx context.Context, statement string, arg ...interface{}) (sql.Result, error) {
+
+	return db.tx.ExecContext(ctx, statement, arg...)
+
+}
+
+func (db *sandboxDB) NamedExecContext(ctx context.Context, statement string, arg interface{}) (sql.Result, error) {
+
+	return db.tx.NamedExecContext(ctx, statement, arg)
+
+}
+
+func (db *sandboxDB) PingContext(ctx context.Context) error {
+
+	return db.db.PingContext(ctx)
+
+}
+
+func (tx *sandboxTx) Commit() error {
+	if tx.rollBackedOrCommitted {
+		return fmt.Errorf("transaction has already been rollbacked or commited")
+	}
+
+	_, err := tx.tx.Exec("RELEASE SAVEPOINT go_pkg_database_sandbox_savepoint;")
```

Comment:
Don't we have to actually Commit the transaction after releasing the savepoint ?

Reply:
nop. `RELEASE SAVEPOINT` is an emulation for the transaction commit. 
Maybe you could find more info here https://www.postgresql.org/docs/8.1/sql-savepoint.html. 
If you still have questions, I'll answer gladly.

Reply:
We don't do actual transaction, we use savepoints as an emulation of transactions. 
The whole sandboxed db is a transaction, if we commited it, it would defeat the purpose.

Reply:
I misunderstood a few things indeed. Thanks for the explanations 🙇